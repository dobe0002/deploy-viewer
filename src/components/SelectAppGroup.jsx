import React from 'react';
import { FormGroup, Label, Input } from 'reactstrap';
import map from 'lodash/map';
import PropTypes from 'prop-types';

export default class SelectAppGroup extends React.Component {
  constructor(props) {
    super(props);
    const { currentAppGroup } = this.props;
    this.buildOptions = this.buildOptions.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.state = {
      selected: currentAppGroup
    };
  }

  buildOptions() {
    const { appGroups } = this.props;
    return map(appGroups, (appDescr, appCode) => {
      return <option value={appCode}>{appDescr}</option>;
    });
  }

  handleChange(event) {
    const { getChange } = this.props;
    const newVal = event.target.value;
    this.setState({ selected: newVal });
    getChange(newVal);
  }

  render() {
    const { selected } = this.state;
    return (
      <FormGroup>
        <Label for='selectAppGroup'>App</Label>
        <Input
          type='select'
          name='select'
          id='selectAppGroup'
          value={selected}
          onChange={this.handleChange}
        >
          <option value=''>All Applications</option>
          {this.buildOptions()}
        </Input>
      </FormGroup>
    );
  }
}
SelectAppGroup.propTypes = {
  appGroups: PropTypes.objectOf,
  currentAppGroup: PropTypes.string,
  getChange: PropTypes.func
};
SelectAppGroup.defaultProps = {
  appGroups: {},
  currentAppGroup: '',
  getChange: () => {}
};
