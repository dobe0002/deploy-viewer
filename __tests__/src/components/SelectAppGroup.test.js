/* eslint-disable react/jsx-filename-extension */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SelectAppGroup from '../../../src/components/SelectAppGroup';

configure({ adapter: new Adapter() });

const { axe, toHaveNoViolations } = require('jest-axe');

expect.extend(toHaveNoViolations);

describe('SelectAppGroup component tests', () => {
  const sampleProps = {
    appGroups: { HR: 'Human Resources', RAS: 'Research' },
    currentAppGroup: ''
  };
  test('Accessibility check', async () => {
    const wrapper = shallow(
      <main>
        <SelectAppGroup appGroups={sampleProps.appGroups} />
      </main>
    ); // Main tag is needed to prevent landmark errors
    const results = await axe(wrapper.html());
    expect(results).toHaveNoViolations();
  });

  test('Snapshot test', () => {
    const wrapper = shallow(
      <SelectAppGroup appGroups={sampleProps.appGroups} />
    );
    expect(wrapper.find('option')).toHaveLength(3);
    expect(wrapper.state('selected')).toEqual('');
    expect(wrapper).toMatchSnapshot();
  });

  test('Sent appGroup is selected', () => {
    const wrapper = shallow(
      <SelectAppGroup appGroups={sampleProps.appGroups} currentAppGroup='RAS' />
    );
    expect(wrapper.find('option')).toHaveLength(3);
    expect(wrapper.state('selected')).toEqual('RAS');
  });
  test('Change selection', () => {
    let mockGetChangeCalled = '';
    const mockGetChange = newAppGroup => {
      mockGetChangeCalled = newAppGroup;
    };
    const wrapper = shallow(
      <SelectAppGroup
        appGroups={sampleProps.appGroups}
        getChange={mockGetChange}
      />
    );
    expect(wrapper.find('option')).toHaveLength(3);
    expect(wrapper.state('selected')).toEqual('');

    wrapper.instance().handleChange({ target: { value: 'HR' } });
    expect(mockGetChangeCalled).toEqual('HR');
    expect(wrapper.state('selected')).toEqual('HR');
  });
});
