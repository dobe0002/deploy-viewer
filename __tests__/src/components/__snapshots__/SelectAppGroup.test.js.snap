// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectAppGroup component tests Snapshot test 1`] = `
<FormGroup
  tag="div"
>
  <Label
    for="selectAppGroup"
    tag="label"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    App
  </Label>
  <Input
    id="selectAppGroup"
    name="select"
    onChange={[Function]}
    type="select"
    value=""
  >
    <option
      value=""
    >
      All Applications
    </option>
    <option
      value="HR"
    >
      Human Resources
    </option>
    <option
      value="RAS"
    >
      Research
    </option>
  </Input>
</FormGroup>
`;
